RFC 0006: Governance Amendment (#47)
QuiltMC/rfcs: structure/0006-governance.md

This document is intended to detail the governing structure of the Quilt project.

# Motivation

> This document has been written with the explicit purpose of defining guidelines and expectations for project staff.

Author: I've considered this, but the 500 character long lines are just not good practice, not even in markdown.
At the minimum I wanted to break after each sentence, but I ended up doing a few more after commas.

It has been written with the idea of decentralizing power in mind, to help contribute to the goal of a truly open project.

# Explanation

## Teams

A team is a group of team members dedicated to solving a specific issue,
maintaining a specific sub project or overseeing a specific area.

> All teams exist as teams on GitHub and can be viewed at https://github.com/orgs/QuiltMC/teams.

Author: yep, same problem as above. Not sure if there's a nice fix for this.

> All teams exist as teams on GitHub and can be viewed at https://github.com/orgs/QuiltMC/teams.

Reviewer: https:\/\/github.com/orgs/QuiltMC/teams/<TEAM>/edit
![image](https://user-images.githubusercontent.com/30361266/155811387-4ebd89ca-80a2-4553-9d76-a52d36ea9e9d.png)

> All teams exist as teams on GitHub and can be viewed at https://github.com/orgs/QuiltMC/teams.

Reviewer 2: ```suggestion
All teams exist as teams on GitHub and can be viewed on our website at https://quiltmc.org/about/teams/.
```

### Team Members

A team member is a member of the organization assigned to a specific team on GitHub.
A member of the organization may be assigned to multiple teams.

> A team must have at least one member.

Author: I don't see a reason to make the lower bound an bigger than one.
A team will be needed for anything that doesn't fit in any other teams. and shouldn't be done by admins.

We also have the infrastructure team with only one member.

### Team Leads

> A team lead is a team member with the Team Maintainer role for the relevant team on GitHub.

Reviewer 3: The Github Docs page ['About team maintainers'](https://docs.github.com/en/organizations/organizing-members-into-teams/assigning-the-team-maintainer-role-to-a-team-member#about-team-maintainers) could be linked here

A member of the organization may be a team lead for multiple teams.
Each team must have at least one team lead.

### Team Responsibilities

Team responsibilities will vary and be detailed on a team-by-team basis.
In general, a team is responsible for performing the tasks outlined in the RFC that proposed its creation.

Most teams will also maintain one or more repositories on GitHub.
The team is responsible for reviewing pull requests and ensuring that they meet the Quilt projects goals in quality and purpose.

> Once a team has decided that a pull request is suitable for merging, a team lead will be responsible for the actual merging.

Author: Ah yes, that one also mentions technical leads. I'll have a look.

> Once a team has decided that a pull request is suitable for merging, a team lead will be responsible for the actual merging.

Reviewer 4: I'd say a note about how a team lead can help guide the general vision and execution of the team's responsibilities may be useful, but perhaps that's not actually what you want from team leads?

> Once a team has decided that a pull request is suitable for merging, a team lead will be responsible for the actual merging.

Reviewer 2: This hasn't always been applied in practice. Each team should define whenever they lock merging to leads or not. I know it isn't in the case of community tooling and infrastructure.
```suggestion
Once a team has decided that a pull request is suitable for merging, a team lead will be responsible for the actual merging. Team leads can also decide to allow some or all team members to perform merges.
```

### Creating a Team

A new team can be created by passing an RFC proposing that team.

> The initial team leads must be specified in the RFC and must approve of the RFC before it is merged.

Author: I guess that would work. I just want to make sure that people can't be made leads without approving of it.
I'll add that tomorrow unless someone has a better idea.

> The initial team leads must be specified in the RFC and must approve of the RFC before it is merged.

Reviewer 5: Considering this on top of RFC-0001 -- is this not a change to the base process itself, requiring a team lead nominate for an RFC to pass?

> The initial team leads must be specified in the RFC and must approve of the RFC before it is merged.

Reviewer 2: That hasn't been applied in practice. The initial team should appear in the PR, but there is no need to add it to the RFC itself. 
```suggestion
The initial team leads must be specified in the pull request adding RFC and must approve of the RFC before it is merged.
```

  - Is there already a team responsible for the described tasks?
  - If this RFC doesn't pass, who would be responsible for the described tasks?

- **Prior Art**:
  - Are there other organizations or projects with comparable tasks?

### Sub Teams

Some teams may wish for more granular organization of their structure.
In those cases, these teams may create sub teams.

Sub teams behave like regular teams, except they are not required to have a team lead.
If the team doesn't have its own team lead, the parent's team leads act as its team lead.
In general, a sub team inherits its parent's structure according to their parent's team RFC.

## The Administrative Board

The administrative board is a group of organization members responsible for tasks that are not assigned to another team.
It is made up of at least three members to prevent deadlocks in the decision-making process.
Members of the administrative board may be members of other teams.

At any point in time, a member of the organization may nominate themselves for a position on the administrative board.
This will then initiate the voting process, with all other organization members being eligible to vote.

> If the nominated member passes a two-thirds vote, they are added to the administrative board.

Author: I don't think forcing an odd number of admins is very feasible.
What if we have 5, and one leaves?
Do we have to elect a replacement immediately?
so we're locked into 5 then? Unless two leave. then we couldn't add another one until we find two new ones.
And I don't think the benefit is that high.

Even if we have an odd number, we can't really guarantee that all of them are around for a tie break, leaving us with an even number of votes anyway.

> If there are less than three members on the administrative board, a special vote is initiated.

Reviewer 6: i'd rephrase "initiated" to "called" to fit in with the wording found under **voting**

Author: Yep, thanks. I changed the wording midway, I'll fix this.

Reviewer 6: any voting surrounding the maintaining of the mandatory minimum of body should have an overal, deterministic deadline by which point (or earlier) new people are assigned to the body.

in my experience, if you leave the overal period indeterminate "as soon as possible", "when they have found a replacement", it can happen that the remaining member(s) drag their feet and it doesn't happen for months.

yes, the alternate election method is already included, but you will find that the majority will have a tendency to not hold this required vote and the few that want it are placated or leveraged to not hold the vote.

by making sure there is always a deadline, this will hopefully make sure the regular members are motivated to hold the vote, even if the remaining body member(s) have been dragging their feet on their required steps.

> Each remaining board member may nominate an organization member.

Reviewer 7: Does this mean that a member can be made a board member against their will?

Organization members may also nominate themselves.
All other organization members may then vote on the nominees.

> The nominee with the most votes will be added to the administrative board.

Reviewer 6: Single Transferable Vote comes highly recommended.

> The nominee with the most votes will be added to the administrative board.

Author: I wouldn't mind a more complex voting system, so if you want to suggest one, go ahead.
Just know that this *needs* (at least) one winner, so a simple majority vote isn't good enough.

> The nominee with the most votes will be added to the administrative board.

Reviewer 8: Yeah, as @BartTerpstra1 mentioned, Single Transferable Vote would be my personal pick here.

> The nominee with the most votes will be added to the administrative board.

Reviewer 6: just so we are clear.
this proposes a multi-candidate election for a single seat, potentially consecutively in if the administrative board has fallen below 2 OR
does this propose a multi-candidate election for up to three seats?

> The nominee with the most votes will be added to the administrative board.

Author: it's a multi-candidate election for up to three seats.
I don't really mind if it's 3 consecutive votes or a single one yielding 3 candidates.
This scenario *should* be very rare.

> The nominee with the most votes will be added to the administrative board.

Reviewer 6: if you'd told me a year ago, i'd agree :p

> The nominee with the most votes will be added to the administrative board.

Reviewer 7: > All other organization members may then vote on the nominees.

With FPTP, I guess this doesn't matter that much, however, with other (better) voting systems, it feels weird to me that you wouldn't allow the nominees to vote.

> In the case of a tie, the remaining board members must break that tie.

Author: Kinda, I've considered this, but it doesn't really make sense here:
- If there are two admins, they either agree on the tie break, or they don't. Doesn't matter if it's worth one or three points.
- If there is one admin, tie break is trivial
- If there is none, you can't break the tie.

> In the case of a tie, the remaining board members must break that tie.

Reviewer 8: Wait, so that means this tie break will still result in a standstill in two out of three scenarios?

> In the case of a tie, the remaining board members must break that tie.

Author: that's why the tie break is not based one a single vote. The admins gotta duke it out somehow.
Realistically, if there's two candidates, they could just accept both.
Theoretically, you could have 20 members tied for first place.

> In the case of a tie, the remaining board members must break that tie.

Reviewer 6: either they duke it out, or you decide at random.
deciding at random actually has historical precedent.
in a single transferable vote tie with actors able to give their true preference, repeating the election doesn't change the result after all.

## Processes

To streamline working with teams, some standard processes need to be defined.
The following processes apply to all teams unless that team's RFC specifies a different process.

### Voting

Unless specified otherwise, the following process is used for voting:

A vote in a team may be called at any time by a team lead for that team or for one of its parent teams.
A member of the administrative board may also initiate a vote in any team. 

When a vote is called within a team, all members of that team, its parent teams, and the administrative board are eligible to vote.
All polls should include a positive and negative option, as well as the option to abstain.

> Voting members that abstain are excluded from percentages and tallies.

Reviewer 9: How does this handle voting members who did not vote within the allotted time frame? Are they counted as having abstained, or separately marked as "not voting"?

Reviewer 2: Right now I believe we count them as abstained, I think that's a good default option

This document is intended to detail the governing structure of the Quilt project.

# Motivation
This document has been written with the explicit purpose of defining guidelines and expectations for project staff.
It has been written with the idea of decentralizing power in mind, to help contribute to the goal of a truly open project.

# Explanation

## Teams

A team is a group of team members dedicated to solving a specific issue,

> maintaining a specific sub project or overseeing a specific area.

Reviewer 10: Use the [Oxford comma](maintaining a specific sub project or overseeing a specific area) to remove ambiguity:
"maintaining a specific sub-project, or overseeing a specific area."

All teams exist as teams on GitHub and can be viewed at https://github.com/orgs/QuiltMC/teams.

### Team Members

A team member is a member of the organization assigned to a specific team on GitHub.
A member of the organization may be assigned to multiple teams.
A team must have at least one member.

### Team Leads

A team lead is a team member with the Team Maintainer role for the relevant team on GitHub.
A member of the organization may be a team lead for multiple teams.

> Each team must have at least one team lead.

Reviewer 4: Moreover, I think something like "A team lead may be nominated via general consensus of the team, or by an election, the process being detailed below." would be good for specificity's sake.

### Team Responsibilities

Team responsibilities will vary and be detailed on a team-by-team basis.
In general, a team is responsible for performing the tasks outlined in the RFC that proposed its creation.

Most teams will also maintain one or more repositories on GitHub.

> The team is responsible for reviewing pull requests and ensuring that they meet the Quilt projects goals in quality and purpose.

Reviewer 11: The correct way to refer to the project is capitalised - so, `The Quilt Project`. Also, you missed an apostrophe - `The Quilt Project's`

Once a team has decided that a pull request is suitable for merging, a team lead will be responsible for the actual merging.

### Creating a Team

A new team can be created by passing an RFC proposing that team.
The initial team leads must be specified in the RFC and must approve of the RFC before it is merged.

> Once the [RFC process](0001-rfc-process.md) has concluded and the RFC was merged,

Reviewer 11: I'd use `is concluded` and `is merged`, or `has concluded` and `has been merged`. Just a nitpick but this reads strangely.

The RFC proposing the new team should contain the following major sections:

- **Motivation**:
  - Why is this team needed?
  - What problems will this team solve?
  - What tasks will this team be responsible for?

- **Explanation**:
  - What will this team *do*?
  - Who will be the team leads?
  - What new projects will be created that this team will be responsible for?
  - How will each of these projects benefit the QuiltMC organization/community?
  - What processes will this team follow for each of the different kinds of changes their projects may undergo?

- **Drawbacks**:
  - Is there overlap of responsibilities?
  - Is there excessive granularity?

- **Rationale and Alternatives**:
  - Is there already a team responsible for the described tasks?
  - If this RFC doesn't pass, who would be responsible for the described tasks?

- **Prior Art**:
  - Are there other organizations or projects with comparable tasks?

### Sub Teams

Some teams may wish for more granular organization of their structure.
In those cases, these teams may create sub teams.

Sub teams behave like regular teams, except they are not required to have a team lead.

> If the team doesn't have its own team lead, the parent's team leads act as its team lead.

Reviewer 10: "If the team doesn't have its..." -> "If the sub team doesn't have its..."

"...parent's team leads act..." -> "...parent team's lead acts..."

Before the vote begins, a group of people, henceforth referred to as the Election Oversight Board or EOB, must be selected to oversee the voting process. They will be responsible for receiving candidate nominations, concerns about nominated candidates from other organization members, and hosting the election. In normal circumstances, the EOB consists of the existing members of the admin board, but, if this is not possible for any reason, the EOB will instead consist of members with the Community Manager role in the Community Team. The initial EOB may add any members that they feel would be a valuable addition to the board, with a simple majority vote.

The vote is conducted using [Single Transferrable Vote](https://en.wikipedia.org/wiki/Single_transferable_vote) (STV), or [Instant Runoff voting](https://en.wikipedia.org/wiki/Instant-runoff_voting) (IRV) in instances where there is only one space on the admin board to be filled, to ensure a minimum amount of "wasted votes". When using Single Transferrable vote, the [Droop Quota](https://en.wikipedia.org/wiki/Droop_quota) must be used to calculate the amount of votes required for a candidate to win a position on the admin board.

> At the start of the voting process, each remaining board member may nominate an organization member, and organization members may also nominate themselves. Nominations must be submitted to the EOB within one week of the start of the process, after which time the vote is opened.

Reviewer 2: We haven't seen any nomination of someone else in practice. This would honestly easily make the person being nominated uncomfortable if they don't want to run for the election, or can be seen as an endorsement otherwise which can spark a conflict of interest. I'd suggest to remove that. 
```suggestion
At the start of the voting process, each organization members may nominate themselves. Nominations must be submitted to the EOB within one week of the start of the process, after which time the vote is opened.
```

> After one week of preparation and nomination submission, nominations must close, the list of candidates must be published, and the vote must begin, lasting one week. The system used to conduct the vote must allow voters to change their votes, and enable voters to rank the candidates in order of preference, while also having the option to not rank any given candidate. All organization members (which at this time is anyone with the "Community Team" or "Quilt Developer" role in Discord) are able to vote. During the vote, voters can contact the EOB about any concerns they have over the vote (for example, concerns about a candidate being elected, or concerns about the vote being influenced by an external force), to be dealt with as outlined in the paragraph below.

Reviewer 7: > All organization members (which at this time is anyone with the "Community Team" or "Quilt Developer" role in Discord) are able to vote.

Are these the people that are members at the start of the vote, or are people that join during the vote also included? What if someone steps down during a vote, or gets removed for certain reasons?

Reviewer 8: The CT's policy is that the list of eligible voters cannot change between the start of the vote and the end of the vote (except if someone comes back from an extended leave, which is irrelevant here). That policy could be adopted here too.

After the voting period concludes, the EOB may take up to one week to count the votes. In the event of a tie, votes by existing admin board members will count as three votes. The EOB may privately inform the winning candidate, before publishing the results to the organization members and subsequently the community at large.

## Processes

To streamline working with teams, some standard processes need to be defined.
The following processes apply to all teams unless that team's RFC specifies a different process.

### Voting

Unless specified otherwise, the following process is used for voting:

A vote in a team may be called at any time by a team lead for that team or for one of its parent teams.

> A member of the administrative board may also initiate a vote in any team. 

Reviewer 2: ```suggestion
A vote can also be initiated as required by another process, such as adding or removing a team member. 
```

> When a vote is called within a team, all members of that team, its parent teams, and the administrative board are eligible to vote.

Reviewer 2: Admins have never been tallied up, nor should they be considering recent events and the discussion that arise from those. 
```suggestion
When a vote is called within a team, all members of that team and its parent teams are eligible to vote.
```

All polls should include a positive and negative option, as well as the option to abstain.
Voting members that abstain are excluded from percentages and tallies.

After one week, the plurality of votes wins.

> ### Removal of Staff

Reviewer 2: We need the process of adding someone to a team written up

Any team member or administrative board member may be removed by a majority vote of all eligible organization members.
The following members are eligible:

> - All members of the relevant team

Reviewer 4: Should we make frequent contributors to the team's GitHub also eligible? A situation like EnnuiL before she actually *joined* the QFAPI team comes to mind.

Reviewer 11: Aren't those people also part of our community?

- All members of all the sub teams of the relevant team
- All members of all the parent teams of the relevant team

> - All administrative board members

Reviewer 2: Same as above
```suggestion
```

role for the relevant team on GitHub.
A member of the organization may be a team lead for multiple teams.
Each team must have at least one team lead.
A team lead may be nominated via general consensus of the team, or by an election, the process being detailed below.

### Team Responsibilities

Team responsibilities will vary and be detailed on a team-by-team basis.
In general, a team is responsible for performing the tasks outlined in the RFC that proposed its creation.

Most teams will also maintain one or more repositories on GitHub.
The team is responsible for reviewing pull requests and ensuring that they meet The Quilt Project's goals in quality and purpose.
Once a team has decided that a pull request is suitable for merging, a team lead will be responsible for the actual merging. Team leads can also decide to allow some or all team members to perform merges.

### Creating a Team

A new team can be created by passing an RFC proposing that team.
The initial team leads must be specified in the pull request adding the RFC and must approve the RFC before it is merged.

> Once the [RFC process](0001-rfc-process.md) is concluded and the RFC us merged,

Reviewer 11: Typo!

a new team with the specified leads will be created on GitHub by the Administrative Board.

The RFC proposing the new team should contain the following major sections:

- **Motivation**:
  - Why is this team needed?
  - What problems will this team solve?
  - What tasks will this team be responsible for?

- **Explanation**:
  - What will this team *do*?

>   - Who will be the team leads?

Reviewer 11: Does this make sense to hardcode into an RFC? I feel like team composition shouldn't require RFC changes.

Author: I think the intent here is to have a ground truth about team composition somewhere.
That being said, it does mention a bit earlier that initial leads should be specified in the PR (rather than the RFC), whch somewhat contradicts this sentiment.

Reviewer 2: The team composition should only be in the PR, I forgot to remove that line

  - What new projects will be created that this team will be responsible for?
  - How will each of these projects benefit the QuiltMC organization/community?
  - What processes will this team follow for each of the different kinds of changes their projects may undergo?

- **Drawbacks**:
  - Is there overlap of responsibilities?
  - Is there excessive granularity?

- **Rationale and Alternatives**:
  - Is there already a team responsible for the described tasks?

>   - If this RFC doesn't pass, who would be responsible for the described tasks?

Reviewer 11: This makes more sense to describe as "which team" rather than "who", so we don't end up hardcoding specific people into the RFCs.

Before the vote begins, a group of people, henceforth referred to as the Election Oversight Board or EOB, must be selected to oversee the voting process. They will be responsible for receiving candidate nominations, concerns about nominated candidates from other organization members, and hosting the election. In normal circumstances, the EOB consists of the existing members of the admin board, but, if this is not possible for any reason, the EOB will instead consist of members with the Community Manager role in the Community Team. The initial EOB may add any members that they feel would be a valuable addition to the board, with a simple majority vote.

The vote is conducted using [Single Transferrable Vote](https://en.wikipedia.org/wiki/Single_transferable_vote) (STV), or [Instant Runoff voting](https://en.wikipedia.org/wiki/Instant-runoff_voting) (IRV) in instances where there is only one space on the admin board to be filled, to ensure a minimum amount of "wasted votes". When using Single Transferrable vote, the [Droop Quota](https://en.wikipedia.org/wiki/Droop_quota) must be used to calculate the amount of votes required for a candidate to win a position on the admin board.

> At the start of the voting process, each organization members may nominate themselves. Nominations must be submitted to the EOB within one week of the start of the process, after which time the vote is opened.

Reviewer 11: This must be a typo - `each organization members`

> After one week of preparation and nomination submission, nominations must close, the list of candidates must be published, and the vote must begin, lasting one week. The list of voters cannot change beyond that point, unless a voter is removed from the organization adn pauses serious threat to the vote integrity. The system used to conduct the vote must allow voters to change their votes, and enable voters to rank the candidates in order of preference, while also having the option to not rank any given candidate. All organization members (which at this time is anyone with the "Community Team" or "Quilt Developer" role in Discord) are able to vote. During the vote, voters can contact the EOB about any concerns they have over the vote (for example, concerns about a candidate being elected, or concerns about the vote being influenced by an external force), to be dealt with as outlined in the paragraph below.

Reviewer 11: I would replace the points here with a bulleted list to aid with reading fatigue, especially given the _massive_ paragraph.

Also, `adn pauses` instead of `and poses`. Are you not using LanguageTool?

`in Discord` should be `on Discord`.

If the EOB receives any concerns over a candidate, either internally or submitted by voter, they must decide if the concerns are worth investigating. If they are not, no further action is taken. If they are, the team must investigate, and, if significant results are found, must publish the results to the voters, so that they can factor them into their decision. The EOB cannot prevent any candidate from running, this is to ensure that they don't have direct control over who can be elected. If, however, the vote is due to end before the EOB can complete their investigation, they may delay the end of the vote by up to 30 additional days.

> After the voting period concludes, the EOB may take up to one week to count the votes. In the event of a tie, votes by existing admin board members will count as three votes. The EOB may privately inform the winning candidate, before publishing the results to the organization members and subsequently the community at large.

Reviewer 11: We do not have a way for these votes to be counted in triplicate because the services we use that handle the voting process make those votes anonymous.

I believe this is an important part of the existing procedure, so this probably needs to be changed.

Author: maybe just having the existing AB vote on the tied candidates would be a better solution?

## Processes

To streamline working with teams, some standard processes need to be defined.
The following processes apply to all teams unless that team's RFC specifies a different process.

### Voting

Unless specified otherwise, the following process is used for voting:

A vote in a team may be called at any time by a team lead for that team or for one of its parent teams.
A vote can also be initiated as required by another process, such as adding or removing a team member. 

When a vote is called within a team, all members of that team and its parent teams are eligible to vote.
All polls should include a positive and negative option, as well as the option to abstain.
Voting members that abstain are excluded from percentages and tallies.
Voting members not casting a vote within the allotted time frame are counted as abstained.

> After one week, the plurality of votes wins. The vote may also be called already if all eligible members 

Reviewer 11: "The plurality of votes" needs replacing, it's not at all clear what this means.

`already` is the wrong term here - maybe `early`?

casted their vote.

### Introduction of Staff

Any community member is eligible to be added to a team by a majority vote of all eligible organization members.

> The vote is started on request from an eligible organization member. A community member wishing to join a team may

Reviewer 11: `by request`, not `on request`.

kindly ask an eligible organization member to request a vote. A positive response from the organization member isn't
mandatory.
The following members are eligible:
- All members of the relevant team
- All members of all the sub teams of the relevant team
- All members of all the parent teams of the relevant team

During the vote, especially if the community member isn't already part of any team, the rest of the organization is
consulted for opinions.

### Removal of Staff

Any team member or administrative board member may be removed by a majority vote of all eligible organization members.
The following members are eligible:
- All members of the relevant team
- All members of all the sub teams of the relevant team

> - All members of all the parent teams of the relevant team

Reviewer 11: It'd probably be good to define this list as "all teams" where the admin board is concerned.

# Drawbacks

In the present day with most accounts and digital spaces being controlled primarily by one person,
the idea of an "administrative board" has some downsides.
A member of the board who is asked to step down may simply refuse to relinquish any resources they control.
For this reason, personal character needs to be a high priority when electing members to the board,

> so that such a situation can be avoided.

Reviewer 11: There are solutions to this, and they should be discussed here, IMO - for example, the infra team controlling ClickUp as it currently does will mean that this can't happen there.

Admins aren't necessarily meant to be resource owners, they're a steering and oversight group - ownership of resources should go to the teams handling them, where that's actually a thing.

Author: With clickup you just move the issue to the infra team instead. That's not *really* a solution.

Having the admin team not being owners completely falls apart in many cases, but the discussion about Admin responsibilities is a separate one that probably doesn't go into this PR (althouh possibly into this RFC).

Once the [RFC process](0001-rfc-process.md) is concluded and the RFC is merged,
a new team with the specified leads will be created on GitHub by the Administrative Board.

The RFC proposing the new team should contain the following major sections:

- **Motivation**:
  - Why is this team needed?
  - What problems will this team solve?
  - What tasks will this team be responsible for?

- **Explanation**:
  - What will this team *do*?
  - What new projects will be created that this team will be responsible for?
  - How will each of these projects benefit the Quilt organization/community?
  - What processes will this team follow for each of the different kinds of changes their projects may undergo?

- **Drawbacks**:
  - Is there overlap of responsibilities?
  - Is there excessive granularity?

- **Rationale and Alternatives**:
  - Is there already a team responsible for the described tasks?
  - If this RFC doesn't pass, which team would be responsible for the described tasks?

- **Prior Art**:
  - Are there other organizations or projects with comparable tasks?

### Sub Teams

Some teams may wish for more granular organization of their structure.

> In those cases, these teams may create sub teams.

Reviewer 9: How are these sub teams created? Is it by an RFC (or an amendement to an existing RFC), or a vote of its parent (with or without a defining document)? Are teams free to have any number of subteams with any level of nesting?

Reviewer 2: As part of the RFC if one exists (you'll notice that a decent amount of historical teams don't have RFCs), nesting can be infinite, there is no need to have a hard limit of them

Sub teams behave like regular teams, except they are not required to have a team lead.
If the sub team doesn't have its own team lead, the parent team lead's acts as its team lead.
In general, a sub team inherits its parent's structure according to their parent's team RFC.

### Triage Teams

A special type of teams called Triage Teams can exist. They are here to provide support to the organization, through tasks
such as triaging and tagging GitHub issues, quality assurance, writing regression tests, etc..

> Members of Triage Teams aren't considered organisation members.

Reviewer 9: What are the consequences of (exclusively) members of triage teams not being considered as members of the organization? Is this effectively akin to being an outside collaborator for a select number of repositories, but they are instead inside the GitHub organization (yet outside of the Quilt organization)?

There was also discussion with regards to whether a member of such a triage team would be considered a Quilt Developer, which I believe led to this additional paragraph. It then begs the question: what is the definition of a Quilt Developer?

## The Administrative Board

> The administrative board is a group of organization members responsible for tasks that are not assigned to another team.

Reviewer 9: Is the administrative board considered a team, as defined above?

Reviewer 2: No, they don't have any team lead

It is made up of at least three members to prevent deadlocks in the decision-making process.
Members of the administrative board may be members of other teams.

> If it is decided that more administrative board members are required, either due to a current member stepping down, or due to it being decided that the admin board needs to be expanded, an election must take place.

Reviewer 9: Who decides when more administrative board members are required? Is it the current administrative board and/or a consensus of the entire Quilt organization?

Reviewer 2: By informal concensus within the organization

> ### Admin board Voting Process

Reviewer 2: I don't think we should be making the wording more ambiguous without any gain. If there is another team needing that process later on then we can rename this section.

 Due to the fact that multiple candidates may be nominated for the admin position when a vote is held, the fact that there may be multiple spots to fill on the board, and the fact that the decision affects the entire organization, the standard voting procedure cannot be used. Instead, the following procedure will be employed:

> Before the vote begins, a group of people, henceforth referred to as the Election Oversight Board or EOB, must be selected to oversee the voting process. They will be responsible for receiving candidate nominations, concerns about nominated candidates from other organization members, and hosting the election. In normal circumstances, the EOB consists of the existing members of the admin board, but, if this is not possible for any reason, the EOB will instead consist of members with the Community Manager role in the Community Team. The initial EOB may add any members that they feel would be a valuable addition to the board, with a simple majority vote.

Reviewer 9: Is there a minimum number of people required to form an Election Oversight Board? Should there be such an enforced minimum, akin to the minimum number of members enforced on the administrative board itself?

Perhaps, as a modicum of accountability and transparency, it should be required that at least one member of the EOB not be an existing member of the administrative board.

Reviewer 2: > As I understand it, the EOB already doesn't require members added later to be from any particular affiliation?

Correct

## Processes

To streamline working with teams, some standard processes need to be defined.
The following processes apply to all teams unless that team's RFC specifies a different process.

### Voting

Unless specified otherwise, the following process is used for voting:

A vote in a team may be called at any time by a team lead for that team or for one of its parent teams.
A vote can also be initiated as required by another process, such as adding or removing a team member. 

When a vote is called within a team, all members of that team and its parent teams are eligible to vote.
All polls should include a positive and negative option, as well as the option to abstain.
Voting members that abstain are excluded from percentages and tallies.
Voting members not casting a vote within the allotted time frame are counted as abstained.

> After one week, the majority wins. The vote may also be called early if all eligible members 

Reviewer 2: No, if all team members but one are inactive we don't want to deadlock a team

casted their vote.

### Introduction of Staff

Any community member is eligible to be added to a team by a majority vote of all eligible organization members.
The vote is started by request from an eligible organization member. A community member wishing to join a team may
kindly ask an eligible organization member to request a vote. A positive response from the organization member isn't

> mandatory.

Reviewer 2: Accepting to hold a vote

The following members are eligible:
- All members of the relevant team
- All members of all the sub teams of the relevant team
- All members of all the parent teams of the relevant team

During the vote, especially if the community member isn't already part of any team, the rest of the organization is
consulted for opinions.

### Removal of Staff

Any team member or administrative board member may be removed by a majority vote of all eligible organization members.
The following members are eligible:
- All members of the relevant team
- All members of all the sub teams of the relevant team
- All members of all the parent teams of the relevant team

> In case the demotion is against an administrative board member, all the existing teams are considered as relevant.

Reviewer 9: It's not a demotion, but a removal.
```suggestion
In case the removal is against an administrative board member, all the existing teams are considered as relevant.
```

### Team Members

A team member is a member of the organization assigned to a specific team on GitHub.
A member of the organization may be assigned to multiple teams.
A team must have at least one member. An individual part of at least one team (except a triage team) is considered an
organization member.

### Team Leads

A team lead is a team member with the [Team Maintainer](https://docs.github.com/en/organizations/organizing-members-into-teams/assigning-the-team-maintainer-role-to-a-team-member#about-team-maintainers) 
role for the relevant team on GitHub.
A member of the organization may be a team lead for multiple teams.
Each team must have at least one team lead.
A team lead may be nominated via general consensus of the team, or by an election, the process being detailed below.

### Team Responsibilities

Team responsibilities will vary and be detailed on a team-by-team basis.
In general, a team is responsible for performing the tasks outlined in the RFC that proposed its creation.

Most teams will also maintain one or more repositories on GitHub.
The team is responsible for reviewing pull requests and ensuring that they meet The Quilt Project's goals in quality and purpose.

> Once a team has decided that a pull request is suitable for merging, a team lead will be responsible for the actual merging. Team leads can also decide to allow some or all team members to perform merges.

Reviewer 2: Yeah, why wouldn't it?

> Once a team has decided that a pull request is suitable for merging, a team lead will be responsible for the actual merging. Team leads can also decide to allow some or all team members to perform merges.

Reviewer 12: fair enough

### Creating a Team

> A new team can be created by passing an RFC proposing that team.

Reviewer 12: proposing -> defining?

if it's passed its not a proposal anymore

- **Rationale and Alternatives**:
  - Is there already a team responsible for the described tasks?
  - If this RFC doesn't pass, which team would be responsible for the described tasks?

- **Prior Art**:
  - Are there other organizations or projects with comparable tasks?

### Sub Teams

Some teams may wish for more granular organization of their structure.
In those cases, these teams may define sub teams in their corresponding RFC.

Sub teams behave like regular teams, except they are not required to have a team lead.
If the sub team doesn't have its own team lead, the parent team lead's acts as its team lead.
In general, a sub team inherits its parent's structure according to their parent's team RFC.

### Triage Teams

A special type of teams called Triage Teams can exist. They are here to provide support to the organization, through tasks

> such as triaging and tagging GitHub issues, quality assurance, writing regression tests, etc..

Reviewer 2: Writing regression tests is an example of what's done by the CPython triage team, which is a big inspiration for this section. Triaging works can also include writing tests, as it gives a minimal reproducible example, and can be incorporated later by the developer who fixes the bug. This is still supporting work.

### Admin Board Election Process

Due to the fact that multiple candidates may be nominated for the admin position when a vote is held, the fact that there may be multiple spots to fill on the board, and the fact that the decision affects the entire organization, the standard voting procedure cannot be used. Instead, the following procedure will be employed:

Before the vote begins, a group of people, henceforth referred to as the Election Oversight Board or EOB, must be selected to oversee the voting process. They will be responsible for receiving candidate nominations, concerns about nominated candidates from other organization members, and hosting the election. In normal circumstances, the EOB consists of the existing members of the admin board, but, if this is not possible for any reason, the EOB will instead consist of members with the Community Manager role in the Community Team. The initial EOB may add any members that they feel would be a valuable addition to the board, with a simple majority vote.

> The vote is conducted using [Single Transferrable Vote](https://en.wikipedia.org/wiki/Single_transferable_vote) (STV), or [Instant Runoff voting](https://en.wikipedia.org/wiki/Instant-runoff_voting) (IRV) in instances where there is only one space on the admin board to be filled, to ensure a minimum amount of "wasted votes". When using Single Transferrable vote, the [Droop Quota](https://en.wikipedia.org/wiki/Droop_quota) must be used to calculate the amount of votes required for a candidate to win a position on the admin board.

Reviewer 9: What happens when the amount of candidates is equal to the amount of open positions being considered in an upcoming election? Do they automatically win those positions and become members of the Administrative Board? 

I'd think there should be a form of confirmatory vote in this situation, so the electorate (the members of the organization) have a chance to discuss the administrative board members-elect and ultimately confirm or deny them their positions, rather than automatically conferring those positions as uncontested.

- All members of all the sub teams of the relevant team
- All members of all the parent teams of the relevant team

In case the removal is against an administrative board member, all the existing teams are considered as relevant.

### Dismissing a Team

If a team is found to be uncapable of managing itself anymore, any team member or administrative board member can,
with enough justification, start a vote to remove all team members from the specified team. All organization members

> are eligible to vote.

Reviewer 11: Another rephrase here.

```
When a team is no longer able to manage itself, any member of that team
or the Administrative Board may, with proper justification, start a vote to
remove all members of the team in question. All organization members
are eligiable to vote in this scenario.
```

If the majority votes in favor, all current team members are removed and the administrative board will designate one 

> or many new team leads who will be in charge of building the new team.

Reviewer 11: `one or more`, not `one or many`. Also, we capitalize `Admin(istrative) Board` in all our other documents.

# Drawbacks

In the present day with most accounts and digital spaces being controlled primarily by one person,
the idea of an "administrative board" has some downsides.
A member of the board who is asked to step down may simply refuse to relinquish any resources they control.
For this reason, personal character needs to be a high priority when electing members to the board,
so that such a situation can be avoided. This can also be avoided by keeping the roles of the administrative board
purely as steering and oversight and delegate ownerships to other relevant teams, such as Infrastructure.

Teams manage their own members, which can lead to a team being corrupted. If that happens, the entire team would

> have to be dismissed.

Reviewer 11: Another rephrase.

```
It's important to consider the possibility that self-managed teams may
become corrupted over time. To handle this scenario, we would need to 
figure out a process to dissolve the entire team.
```
